frontend/src/features/projects/common/ProjectRouter.tsx

```diff
@@ -32,33 +34,26 @@ const ProjectRouter = ({ location }: { match: Match; location: Location }) => {
   const query = location?.search ?? {};
   const formikRef = useRef<FormikValues>();
   const projectNumber = queryString.parse(query).projectNumber;
+  const { project } = useProject();
   const dispatch = useDispatch();
   const getProjectRequest = useSelector<RootState, IGenericNetworkAction>(
-    state => (state.network as any)[ProjectActions.GET_PROJECT] as any,
+    state => (state.network as any)[ProjectActions.GET_PROJECT_WORKFLOW] as any,
   );
-  const project: any = useSelector<RootState, IProjectWrapper>(state => state.project).project;
   useEffect(() => {
     if (projectNumber !== null && projectNumber !== undefined) {
       dispatch(clearProject());
-      dispatch(fetchProject(projectNumber as string));
+      (dispatch(fetchProject(projectNumber as string)) as any).then((project: IProject) => {
+        dispatch(fetchProjectWorkflow(project?.workflowCode));
```

Comment:
We only load (and store in redux) the current workflow based on project status. As a result we need to load the workflow dynamically based on the project status before displaying a component to ensure consistent behaviour.